### PR TITLE
Recursive

### PR DESCRIPTION
**Pull request by ignasirv**
- Add recursivity
- Set STATE_ROOT_STORAGE_POS to 1

### CODE REVIEW

File: main/main.zkasm [hunk 1]
```diff
@@ -45,20 +46,20 @@ start: ; main zkROM entry point
...
+        %GLOBAL_EXIT_ROOT_STORAGE_POS           :HASHK(E) ; Storage position of the global exit root map
+        HASHPOS                                 :HASHKLEN(E)
+        $ => C                                  :HASHKDIGEST(E)
 
         %ADDRESS_GLOBAL_EXIT_ROOT_MANAGER_L2 => A
         %SMT_KEY_SC_STORAGE => B
-        $ => D                              :MLOAD(numBatch)
-        $ => SR                             :SSTORE ; Store 'numBatch' in storage position 'keccak256(globalExitRoot, 0)'
+        $ => D                              :MLOAD(newNumBatch)
+        $ => SR                             :SSTORE ; Store 'newNumBatch' in storage position 'keccak256(globalExitRoot, 0)'
```
Comment (invocamanman) [inline main/main.zkasm:62]: globalExitRoot, numBatch

File: main/main.zkasm [hunk 2]
```diff
@@ -2,39 +2,40 @@ INCLUDE "constants.zkasm"
...
+
+        ; Fill newAccHashInput: oldAccInputHash
         32 => D
-        $ => A                              :MLOAD(oldStateRoot)
-        A                                   :HASHK(0) ; GlobalHash address is 0
-        A => SR
-        $ => A                              :MLOAD(oldLocalExitRoot)
-        A                                   :HASHK(0)
+        $ => A                              :MLOAD(oldAccInputHash)
+        A                                   :HASHK(0) ; 'newAccHashInput' hashK address is 0
```
Comment (invocamanman) [inline main/main.zkasm:33]: This parameter is in C, no need to load it

Comment (krlosMata) [reply main/main.zkasm:33]: I think it is better to compute all the `newAccHashInput` at the end for clarity. So this comment does not longer apply.

[the remark this answers is not in the document]
Comment (krlosMata) [reply main/main.zkasm:33]: agree !!

File: main/main.zkasm [hunk 3]
```diff
@@ -133,82 +135,77 @@ processTxFinished:
...
-
 
 ;;;;;;;;;;;;;;;;;;
 ;; F - Finalize execution
 ;;;;;;;;;;;;;;;;;;
-finalizeExecution:
-        0 => A,B,C,D,E,CTX, SP, PC, GAS, MAXMEM, SR, HASHPOS, RR ; Set all registers to 0
-                                                                        :JMP(finalWait)
+        ; Set output registers (SR register is the 'newStateRoot')
+        SR                              :MSTORE(newStateRoot)
```
[the remark this answers is not in the document]
Comment (krlosMata) [reply main/main.zkasm:197]: It is just for clarity to be used later on in the full-tracer and in the executor (`assertOutputs` function)

**Review by krlosMata — APPROVED** [review]
:smile_cat:

File: main/main.zkasm (hunk 3 above)
[the remark this answers is not in the document]
Comment (invocamanman) [reply main/main.zkasm:193]: i think i would not load things in memory just for the tracer, we can call a pure execution function to do the trick `$ or $$`

**Review by invocamanman — APPROVED** [review]
^^

**Pull request merged**